GiniSwitchSDK/Example/GiniTariffSDK/ExtractionsViewController.swift

```diff
@@ -62,7 +62,15 @@ extension ExtractionsViewController: UITableViewDataSource {
         cell.nameLabel.text = extraction?.name ?? ""
         cell.valueTextField.text = extraction?.valueString ?? ""
         cell.onExtractionChange = { (newValue: String) -> Void in
-            extraction?.value = ExtractionValue(value: newValue as AnyObject, unit: extraction?.value.unit)
+            // Since the data is shown in a text field, all entries are strings. However, it is
+            // important to NOT change the data type in the extraction collection
+            // (otherwise the backend might reject the data)
+            var replacementValue:Any = newValue
+            let doubleValue = Double(newValue)
+            if let doubleValue = doubleValue {
+                replacementValue = doubleValue
+            }
+            extraction?.value = ExtractionValue(value: replacementValue, unit: extraction?.value.unit)
```

Comment:
Having the explanation above, I think this could be simplified and still be readable.

`extraction?.value = ExtractionValue(value: Double(newValue) ?? newValue, unit: extraction?.value.unit)
`

What do you think?

Reply:
Good point, yeah.

Reply:
Good point, yeah.